src/Queue/RabbitMQQueue.php

```diff
@@ -123,11 +129,18 @@ public function pop($queueName = null)
     {
         try {
             /** @var AmqpQueue $queue */
-            list($queue) = $this->declareEverything($queueName);
+            list($queue) = $this->declareEverythingOnce($queueName);
 
-            $consumer = $this->context->createConsumer($queue);
+            // create the consumer once and cache it
+            $consumer = $this->createConsumerOnce($queue);
 
-            if ($message = $consumer->receiveNoWait()) {
+            if (isset($this->receiveConfig['method']) and $this->receiveConfig['method'] == 'basic_consume') {
+                $message = $consumer->receive($this->receiveConfig['timeout']);
```

Comment:
fyi: the method using basic get under the hood in 0.9. To use basic consume a subscription consumer should be used

Reply:
thanks, I'll read more docs about this